### ISSUE TITLE
SegFormer

### ISSUE THREAD

**Issue by chefkrym** [qubvel/segmentation_models.pytorch#762]
Kindly requesting you add segformer please :)

**Comment by rishiswethan** [comment]
https://github.com/lucidrains/segformer-pytorch

A simple pip install from this repo solved this for me. You can simply use,

```
import torch
from segformer_pytorch import Segformer

model = Segformer(
    dims = (32, 64, 160, 256),      # dimensions of each stage
    heads = (1, 2, 5, 8),           # heads of each stage
    ff_expansion = (8, 8, 4, 4),    # feedforward expansion factor of each stage
    reduction_ratio = (8, 4, 2, 1), # reduction ratio of each stage for efficient attention
    num_layers = 2,                 # num layers of each stage
    decoder_dim = 256,              # decoder dimension
    num_classes = 4                 # number of segmentation classes
)
```



instead of,

```
model = smp.DeepLabV3Plus(
            encoder_name=
            encoder_weights=,
            classes=,
            activation=,
        )
```

**Comment by chefkrym** [comment]
> https://github.com/lucidrains/segformer-pytorch
> 
> A simple pip install from this repo solved this for me. You can simply use,
> 
> ```
> import torch
> from segformer_pytorch import Segformer
> 
> model = Segformer(
>     dims = (32, 64, 160, 256),      # dimensions of each stage
>     heads = (1, 2, 5, 8),           # heads of each stage
>     ff_expansion = (8, 8, 4, 4),    # feedforward expansion factor of each stage
>     reduction_ratio = (8, 4, 2, 1), # reduction ratio of each stage for efficient attention
>     num_layers = 2,                 # num layers of each stage
>     decoder_dim = 256,              # decoder dimension
>     num_classes = 4                 # number of segmentation classes
> )
> ```
> 
> instead of,
> 
> ```
> model = smp.DeepLabV3Plus(
>             encoder_name=
>             encoder_weights=,
>             classes=,
>             activation=,
>         )
> ```
> 
> If you already have your generators, loss functions, training pipeline etc, you still keep them. You can also add many other such models this way

Thank you SO SO much @rishiswethan

**Comment by chefkrym** [comment]
> > https://github.com/lucidrains/segformer-pytorch
> > A simple pip install from this repo solved this for me. You can simply use,
> > ```
> > import torch
> > from segformer_pytorch import Segformer
> > 
> > model = Segformer(
> >     dims = (32, 64, 160, 256),      # dimensions of each stage
> >     heads = (1, 2, 5, 8),           # heads of each stage
> >     ff_expansion = (8, 8, 4, 4),    # feedforward expansion factor of each stage
> >     reduction_ratio = (8, 4, 2, 1), # reduction ratio of each stage for efficient attention
> >     num_layers = 2,                 # num layers of each stage
> >     decoder_dim = 256,              # decoder dimension
> >     num_classes = 4                 # number of segmentation classes
> > )
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > instead of,
> > ```
> > model = smp.DeepLabV3Plus(
> >             encoder_name=
> >             encoder_weights=,
> >             classes=,
> >             activation=,
> >         )
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > If you already have your generators, loss functions, training pipeline etc, you still keep them. You can also add many other such models this way
> 
> Thank you SO SO much @rishiswethan

@rishiswethan Have you tried it with a pretrained encoder like they do here? (e.g. using a Resnet etc). Thank you.
